pulp_rpm/tests/functional/api/test_acs.py

```diff
@@ -19,94 +9,70 @@
-        acs_remote = self.remote_api.create(gen_rpm_remote(url=acs_url, policy="on_demand"))
-        self.addCleanup(self.remote_api.delete, acs_remote.pulp_href)
-
-        acs_data = {
-            "name": "alternatecontentsource",
-            "remote": acs_remote.pulp_href,
-            "paths": paths,
-        }
-        acs = self.acs_api.create(acs_data)
-        self.addCleanup(self.acs_api.delete, acs.pulp_href)
-
-        repo = self.repo_api.create(gen_repo())
-        self.addCleanup(self.repo_api.delete, repo.pulp_href)
-        remote = self.remote_api.create(gen_rpm_remote(url=remote_url))
-        self.addCleanup(self.remote_api.delete, remote.pulp_href)
-
-        # Sync repo with metadata only, before ACS refresh it should fail
-        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
-
-        with self.assertRaises(PulpTaskError) as ctx:
-            sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
-            monitor_task(sync_response.task)
-
-        self.assertIn("404, message='Not Found'", ctx.exception.task.error["description"])
-
-        # ACS refresh
-        acs_refresh = self.acs_api.refresh(acs.pulp_href)
-        monitor_task_group(acs_refresh.task_group)
-
-        # Sync repository with metadata only
-        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+@pytest.mark.parametrize(
+    "paths,remote_url,content_summary",
+    [
+        (["rpm-unsigned/"], RPM_ONLY_METADATA_REPO_URL, RPM_FIXTURE_SUMMARY),
+        (
+            ["rpm-distribution-tree/"],
+            RPM_KICKSTART_ONLY_META_FIXTURE_URL,
+            RPM_KICKSTART_FIXTURE_SUMMARY,
+        ),
+    ],
+)
+def test_acs_simple(
+    paths,
+    remote_url,
+    content_summary,
+    rpm_repository_version_api,
+    rpm_repository_api,
+    rpm_acs_api,
+    rpm_repository_factory,
+    rpm_rpmremote_factory,
+    monitor_task,
+    monitor_task_group,
+    gen_object_with_cleanup,
+    delete_orphans_pre,
+):
+    """Test to sync repo with use of ACS."""
+    # ACS is rpm-unsigned repository which has all packages needed
+    acs_remote = rpm_rpmremote_factory(url=PULP_FIXTURES_BASE_URL, policy="on_demand")
+
+    acs_data = {
+        "name": "alternatecontentsource",
+        "remote": acs_remote.pulp_href,
+        "paths": paths,
+    }
+    acs = gen_object_with_cleanup(rpm_acs_api, acs_data)
+
+    repo = rpm_repository_factory()
+    remote = rpm_rpmremote_factory(url=remote_url)
+
+    # Sync repo with metadata only, before ACS refresh it should fail
+    repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+
+    with pytest.raises(PulpTaskError) as ctx:
+        sync_response = rpm_repository_api.sync(repo.pulp_href, repository_sync_data)
         monitor_task(sync_response.task)
 
-        return self.repo_api.read(repo.pulp_href)
+    assert "404, message='Not Found'" in ctx.value.task.error["description"]
```

Comment:
Optional change since it was already like this - is there a way we can just test the error code?  Doing it by string like this seems, not fragile exactly but not entirely proper.

Reply:
Otherwise this looks good

Reply:
Don't think so, this error comes from the task failing. The sync task will encounter a 404 and fail to finish and we are checking against the task's error message. Unless you want me to shorten the string to only check for "404", I don't see a better way.

Reply:
Ok.  Sounds fine then